bench/wtperf/config.c

```diff
@@ -653,6 +666,43 @@ config_sanity(CONFIG *cfg)
 }
 
 /*
+ * config_to_file --
+ *	Write the final config used in this execution to a file.
+ */
+void
+config_to_file(CONFIG *cfg)
+{
+	CONFIG_QUEUE_ENTRY *config_line;
+	FILE *fp;
+	size_t req_len;
+	char *path;
+
+	/* Backup the config */
+	req_len = strlen(cfg->home) + 100;
+	if ((path = calloc(req_len, 1)) == NULL) {
+		(void)enomem(cfg);
+		return;
+	}
+
+	snprintf(path, req_len + 14, "%s/CONFIGBACKUP", cfg->home);
+	if ((fp = fopen(path, "w")) == NULL) {
+		lprintf(cfg, errno, 0, "%s", path);
+		return;
+	}
+
+	/* Print the config dump */
+	while (!TAILQ_EMPTY(&cfg->config_head)) {
+		config_line = TAILQ_FIRST(&cfg->config_head);
+		TAILQ_REMOVE(&cfg->config_head, config_line, c);
+		fprintf(fp, "%s\n", config_line->string);
+		free(config_line->string);
```

Comment:
There is a shared function called ``config_free`` that is a better place for the cleanup. That way it will be cleaned up even if we don't take this code path, and is more consistent with the rest of the config structure.